client/src/components/App/Content/InvestigationForm/InvestigationForm.tsx

```diff
@@ -81,39 +68,66 @@ const InvestigationForm: React.FC = (): JSX.Element => {
             setSymptomsStartDate, setExposureDate, setHasSymptoms, setEndInvestigationDate]
     );
 
-    const { currentTab, confirmFinishInvestigation, handleSwitchTab, saveCurrentTab, isButtonDisabled, areThereContacts } = useInvestigationForm({ clinicalDetailsVariables, personalInfoData, exposuresAndFlightsVariables, interactedContactsState });
-    
-    const shouldDisableButton = isButtonDisabled(currentTab.name);
-    const isLastTab = (currentTab.id === LAST_TAB_ID || (currentTab.id === LAST_TAB_ID - 1 && !areThereContacts));
-    
+    const { confirmFinishInvestigation, areThereContacts } = useInvestigationForm();
+    const {
+        currentTab,
+        moveToNextTab,
+        setNextTab
+    } = useTabManagement();
+
+    const isInvestigationValid = () => {
+        return !(formsValidations.some((formValidation) => !formValidation));
+    }
+
+    const handleClick = () => {
+        if(currentTab === LAST_TAB_ID) { 
```

Comment:
Isn't currentTab === getLastTabId() the check you need?

Reply:
you are right, done.